ruth/tools/simulator.py

```diff
@@ -83,14 +83,17 @@ def single_node_simulator(ctx,
                           k_alternatives,
                           nproc,
                           out,
-                          seed,
                           walltime_s,
+                          seed,
                           continue_from):
     # ensure that ctx.obj exists and is a dict (in case `cli()` is called by means other than the `if` block bellow)
     ctx.ensure_object(dict)
 
     ctx.obj['DEBUG'] = debug
-    walltime = timedelta(seconds=walltime_s) if walltime_s is not None else None
+    if  walltime_s is not None and walltime_s > 0:
+        walltime = timedelta(seconds=walltime_s)
+    else:
+        exit("Walltime '--walltime-s' is necessary and must be a positive integer.")
```

Comment:
Please make it still optional (sorry for bat explanation). The inner walltime should be ignored if no walltime is provided.